### ISSUE TITLE
Array of images bug

### ISSUE THREAD

**Issue by monday1994** [mozilla-services/react-jsonschema-form#797]
### Prerequisites

- [x] I have read the [documentation](https://github.com/mozilla-services/react-jsonschema-form/blob/master/README.md#readme);
- [x] In the case of a bug report, I understand that providing a [SSCCE](http://sscce.org/) example is tremendously useful to the maintainers.

### Description

I've got array of objects, where each object is a pair of {imageName : 'sth', file : 'base64'}.
Example problem flow:
1. I add three objects - [{imageName1 : 'sth', file1 : 'first base64'}, {imageName2 : 'sth2', file2 : 'second base64'},{imageName3 : 'sth3', file3 : 'third base64'}]
2. All three are displayed normally (name and picture below)
3. When I try to remove (by clicking default red button with 'x') second object then second object name is being removed properly but it image not.
4. When I submit this data it is uploaded properly (correct image and name is in second object)

Another way of explanation
1. [{imageName1 : 'sth', file1 : 'first base64'}, {imageName2 : 'sth2', file2 : 'second base64'},{imageName3 : 'sth3', file3 : 'third base64'}]
2. Second object remove - [{imageName1 : 'sth', file1 : 'first base64'}, {imageName3 : 'sth3', file3 : 'second base64'}]
3. After submit data is correct - [{imageName1 : 'sth', file1 : 'first base64'}, {imageName3 : 'sth3', file3 : 'third base64'}]

Summing up above - submitted data is correct but displayed data not.

### Version

1.0.0

**Comment by edi9999** [comment]
Please provide a reproducible example. From the text that you have written, I have no clue as to how to reproduce your issue.
